dataland-e2etests/src/test/kotlin/org/dataland/e2etests/TestDataProvider.kt

```diff
@@ -0,0 +1,83 @@
+package org.dataland.e2etests
+
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.ToJson
+import com.squareup.moshi.Types
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import org.dataland.datalandbackend.openApiClient.model.CompanyAssociatedDataEuTaxonomyData
+import org.dataland.datalandbackend.openApiClient.model.CompanyInformation
+import org.dataland.datalandbackend.openApiClient.model.EuTaxonomyData
+import org.springframework.core.io.ClassPathResource
+import java.lang.reflect.ParameterizedType
+import java.math.BigDecimal
+import java.time.LocalDate
+
+object BigDecimalAdapter {
+    @FromJson
+    fun fromJson(string: String) = BigDecimal(string)
+
+    @ToJson
+    fun toJson(value: BigDecimal) = value.toString()
+}
+
+object LocalDateAdapter {
+    @FromJson
+    fun fromJson(string: String) = LocalDate.parse(string)
+
+    @ToJson
+    fun toJson(value: LocalDate) = value.toString()
+}
+
+class TestDataProvider {
+    private var dataCounter = 0
+
+    private val companyJson = ClassPathResource("/CompanyInformation.json").file
+    private val dataJson = ClassPathResource("/CompanyAssociatedEuTaxonomyData.json").file
+    private val companyJsonString = companyJson.inputStream().bufferedReader().readText()
+    private val dataJsonString = dataJson.inputStream().bufferedReader().readText()
+
+    private val moshi: Moshi = Moshi.Builder().add(KotlinJsonAdapterFactory())
+        .add(BigDecimalAdapter).add(LocalDateAdapter).build()
+    private val companyType: ParameterizedType = Types
+        .newParameterizedType(List::class.java, CompanyInformation::class.java)
+    private val dataType: ParameterizedType = Types
+        .newParameterizedType(List::class.java, CompanyAssociatedDataEuTaxonomyData::class.java)
+    private val companyJsonAdapter: JsonAdapter<List<CompanyInformation>> = moshi.adapter(companyType)
+    private val dataJsonAdapter: JsonAdapter<List<CompanyAssociatedDataEuTaxonomyData>> = moshi.adapter(dataType)
+
```

Comment:
theoretisch kann man comapnies und data random erzeugen. aber das machen wir nicht weil sonar merckrt wenn es gibt randome Daten. Deswegen erzeugen wir randome Daten durch fixture faker und nutzen wir es hier. 
Aber das wird später  weg wenn csv-parser kommt und daher finde ich es erstmal ok

Reply:
Ja der faker erzeugt zwar auch random data, aber wir lassen ihn ja nur zu bestimmten Zeitpunkten manuell laufen.
Wir lassen ihn nur laufen, wenn wir (aus Gründen) neue Test-Daten brauchen. Die Test-Daten werden also nicht bei jedem Test-Run neu erzeugt.
Wenn wir vorher z.B. in unserem DummyDataCreator random-Funktionen hatten, dann liefen die jedes mal, wenn der DummyDataCreator genutzt wird, und das passierte bei jedem Test-Run.
Das ist ein Unterschied zum jetzigen Vorgehen.